### PR TITLE
Add \xHH Unicode escape code to basic strings

### PR DESCRIPTION
**Pull request by eksortso**
Per a conversation in #622, this PR adds a two-character \x escape code sequence for the first 256 Unicode code points. This provides a short explicit sequence for, among other code points: ASCII characters, certain control characters, and whitespace characters such as tab (\x09) and space (\x20) that need visible representation without resorting to longer \u and \U codes. Such sequences are commonly found in languages that also support the \u and \U codes for Unicode.

This PR replaces #709 and is intended for a post-v1.0.0 version of the TOML standard.

### CODE REVIEW

**Comment by marzer** [comment]
@pradyunsg Where did we land with this? It seems like a friction-free addition to the language (and one that would bring TOML in line with many other languages which also support `\x`). It's also practically free for implementers since the code paths that would already be handling `\u` and `\U` can be [trivially extended](https://github.com/marzer/tomlplusplus/blob/16d7ba84803827d08577ea95dd9c86a832878fa4/include/toml%2B%2B/impl/parser.inl#L1399-L1406) to support `\x`.

**Comment by pradyunsg** [comment]
Yea, let's do this.

**Comment by eksortso** [comment]
I'll update it later this evening.

**Comment by marzer** [comment]
Awesome, thanks @eksortso

**Comment by eksortso** [comment]
@marzer I think that's got it.

**Comment by eksortso** [comment]
The second change to `CHANGELOG.md` dresses up the bare URL for toml.io, just to appease the linter. I'll undo it if necessary.

**Comment by eksortso** [comment]
For some reason I'd thought this had already been taken care of. It's up to date again.

@marzer @pradyunsg Care to take look at it?

**Review by marzer — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,7 +21,8 @@
 
 ## 1.0.0-rc.2 / 2020-08-09
 
-* Create https://toml.io as the new primary location to read the TOML spec.
+* Create [https://toml.io](https://toml.io) as the new primary location to read
+  the TOML spec.
```
Comment (ChristianSi) [inline CHANGELOG.md:25]: Does this belong here? It feels unrelated.

**Review by ChristianSi — CHANGES_REQUESTED** [review]
Looks good otherwise!

File: CHANGELOG.md (hunk 1 above)
Comment (eksortso) [reply CHANGELOG.md:25]: Thhe markdown linter that I use is allergic to bare URLs, it seems. I'll undo that bit.

**Review by ChristianSi — APPROVED** [review]

**Comment by eksortso** [comment]
@pradyunsg Any updates on this? Is the issue with `pre-commit.ci` impeding merges? Or are you planning a larger roadmap for the next release of TOML?

**Review by pradyunsg — APPROVED** [review]

**Comment by pradyunsg** [comment]
Thanks @eksortso! ^>^
